### PR TITLE
various AWS,GCP 7.x changes and aws data-voters

### PR DESCRIPTION
**Pull request by shaigbdb**
In this PR:

yml definitions, ES api calls (mostly) conform to 8.x
data nodes are now hot-content (h+c) rather than data (d). This helps add tiering to the cluster.
heap dumps are disabled - those usually just exploded disks out of nowhere.
conformed to new GCP API filter syntax for unattached volume check
improved packer gcp image naming and added manifests
GCP and AWS nodes now on ubuntu 22.04
grafana installation removed. I've left in optional elasticsearch monitoring in case someone still wants it, but didn't check whether it works
added encryption keys for security, reports. saved objects isn't available for all 7.x versions and will be added in 8.x
moved es password in Kibana to keystore
AWS deployment conforms to new terraform
data-voters in both gcp and aws
limited aws vpc endpoint ports 
added debug bootstrap mode and various logs
removed google-beta usages
added tfvars examples

known issues:
Azure untouched and probably not working
1-master clusters don't get bootstrapped well (but 1-master + 2 data voters do, and 3-master clusters work well). I suspect that was an existing issue.

### CODE REVIEW

File: assets/scripts/common/config-es.sh [hunk 1]
```diff
@@ -101,6 +101,9 @@ if [ "$use_g1gc" = "true" ]; then
   sudo sed -i 's/[0-9]\+-:-XX:InitiatingHeapOccupancyPercent/10-:-XX:InitiatingHeapOccupancyPercent/ig' /etc/elasticsearch/jvm.options
 fi
 
+# Disable heap dumps
+echo "-XX:-HeapDumpOnOutOfMemoryError" | sudo tee -a /etc/elasticsearch/jvm.options
```
Comment (LiorF-BDBQ) [inline assets/scripts/common/config-es.sh:105]: should be in jvm.options.d in anotehr file

File: terraform-aws/variables.tf [hunk 2]
```diff
@@ -204,5 +210,10 @@ variable "log_size" {
 
 variable "log_level" {
   description = "log4j log level"
-  default     = "WARN"
-}
\ No newline at end of file
+  default     = "INFO"
+}
+
+variable "debug_bootstrap" {
```
Comment (LiorF-BDBQ) [inline terraform-aws/variables.tf:216]: I think `auto_shut_down_bootstrap_node` is a better var name

**Review by LiorF-BDBQ — COMMENTED** [review]
Looks good, couple of minor comments

File: assets/scripts/common/config-es.sh (hunk 1 above)
Comment (shaigbdb) [reply assets/scripts/common/config-es.sh:105]: I assume that would also be true for the g1gc parameters?
```

# Setup GC
if [ "$use_g1gc" = "true" ]; then
  sudo sed -i -re 's/# ([0-9]+-[0-9]+:-XX:-UseConcMarkSweepGC)/\1/ig' /etc/elasticsearch/jvm.options
  sudo sed -i -re 's/# ([0-9]+-[0-9]+:-XX:-UseCMSInitiatingOccupancyOnly)/\1/ig' /etc/elasticsearch/jvm.options
  sudo sed -i 's/[0-9]\+-:-XX:+UseG1GC/10-:-XX:+UseG1GC/ig' /etc/elasticsearch/jvm.options
  sudo sed -i 's/[0-9]\+-:-XX:G1ReservePercent/10-:-XX:G1ReservePercent/ig' /etc/elasticsearch/jvm.options
  sudo sed -i 's/[0-9]\+-:-XX:InitiatingHeapOccupancyPercent/10-:-XX:InitiatingHeapOccupancyPercent/ig' /etc/elasticsearch/jvm.options
fi
```

File: terraform-aws/variables.tf (hunk 2 above)
Comment (shaigbdb) [reply terraform-aws/variables.tf]: fixed

File: assets/scripts/common/config-es.sh (hunk 1 above)
Comment (shaigbdb) [reply assets/scripts/common/config-es.sh]: Fixed for heapdump
not yet fixed for g1gc, if needed

Comment (LiorF-BDBQ) [reply assets/scripts/common/config-es.sh]: I'm not sure we need this flag at all now that elastic ships with a built-in jvm 17.

Comment (shaigbdb) [reply assets/scripts/common/config-es.sh]: relevant for 7.6 and below. Documented this in the variable description

**Pull request merged**